tests/integration/redis_cluster_tests.py

```diff
@@ -156,3 +157,12 @@ def test_pipeline(self):
         self.assertTrue(span_observer.on_start_called)
         self.assertTrue(span_observer.on_finish_called)
         self.assertIsNone(span_observer.on_finish_exc_info)
+
+        labels = {
+            "redis_client_name": "",
+            "redis_type": "standalone",
+            "redis_database": "0",
+            "redis_command": "GET",
+            "redis_success": "true",
+        }
+        assert REGISTRY.get_sample_value("redis_client_requests_total", labels) == 5.0
```

Comment:
Why test only the gets?

I feel like the test here should spell out what we expect to track, at minimum for clarity purposes:

total = 8
gets = 5
sets = 1
delete = 1
pipeline = 1

Reply:
Aren't we then just testing the redis library? Every command follows the exact same code path in baseplate. It's passed through redis [pretty much always the same way](https://github.com/redis/redis-py/blob/031b2087b57713d29b98140fb1f0b9f7b4fc7a21/redis/commands/core.py#L1410-L1435), and back to our mocked connection object. 

Happy to add it, but would have appreciate this kind of feedback up front.

Reply:
I can't provide feedback on code that hasn't been written yet :)

To be clear, what I'm calling out here is that you changed metrics, did a bunch of things to work through the pipeline code you wrote (which is great), but then only tested part of it. Tests should capture every aspect that we want to track, and in this case, it was a miss from the pipeline tracking. You got a test for tracking the gets within the pipeline, but not the pipeline tracking itself (which would fail because you never increment the total count for pipeline calls, nor are you tracking inc or dec for the active call) which would have been covered if writing assertions for the behavior that the test is performing (that was my original comment [here](https://github.com/reddit/baseplate.py/pull/732#discussion_r945893821) )

I'm not totally clear myself on what we want, which is partially why I'm asking for the tests that cover all the metrics we expect to be emitted from the test case here. Defining it as tests makes it much clearer than is defined elsewhere and easier to validate. Not defining it leads to ambiguity and potential for change without realizing that it's not fully covered by the unit test.

Reply:
TBH, this shouldn't be tacked onto an existing test, you should specifically have a test for metrics separate from any other test so that it's apparent what we're testing and what's not being tested.